script.js

```diff
@@ -0,0 +1,151 @@
+    let row = document.querySelector('#row').value
+    let gopher = document.querySelector('#gopher').value
+    let parsedRow = parseInt(row)
+    let parsedColumn = parseInt(column)
+    let parsedGopher = parseInt(gopher)
+    let parsedObject = {
+        row: parsedRow,
+        column: parsedColumn,
+        gopher: parsedGopher
+    }
+    if (parsedGopher > Math.floor((parsedColumn * parsedRow) / 2)) {
+        document.querySelector(".error_container").textContent = "You must choose a number of gophers between 3 and "
+            + Math.floor((parsedColumn * parsedRow) / 2)
+        return false
+    }
+    Object.values(parsedObject).forEach(function (input) {
+        if (!Number.isInteger(input)) {
+            document.querySelector(".error_container").textContent = "What you playing at sucka! (incorrect datatype)"
+            return false
+        }
+    })
+    if (!checkRange(parsedObject.column)) {
+        document.querySelector(".error_container").textContent = "Please enter a value between 3 and 12!"
+        return false
+    }
+    if (!checkRange(parsedObject.row)) {
+        document.querySelector(".error_container").textContent = "Please enter a value between 3 and 12!"
+        return false
+    }
+
+    return parsedObject
+}
+
+/** Randomly selects a number of cells to change to hit class
+ *
+ * @param height inputted height
+ * @param width inputted width
+ * @param hits inputted desired number of hits
+ */
+function hitGenerator(height, width, hits){
+    let gridSize = height * width
+    if (hits > gridSize) return // Checks if hits exceeds the size of the grid, and kicks out of function if it is
+    let targetCells = [] // Keeps track of the cells that have already been turned to 'hit'
+    for (let i = 0; i < hits; i++){
+        let targetCell = Math.floor(Math.random() * (gridSize)) + 1  //generates a random number between (inclusive of) 1 and hits
+        if (!targetCells.includes(targetCell)){  // Checks whether the cell has been targeted before
+            targetCells.push(targetCell)         // Adds the cell to the array of targeted cells
+            document.querySelector(`[data-grid-id = "${targetCell}"]`).classList.add('hit')
+            document.querySelector(`[data-grid-id = "${targetCell}"]`).classList.remove('miss')
+        } else {
+            i--  // Decrements the counter if failed to find new cell (repeated cell), so that correct number of cells become hits
+        }
+    }
+}
+
+
+function toggleDisplay() {
+    splash.classList.toggle('hidden')
+    gameBoard.classList.toggle('hidden')
+}
+
+/** Start a new game with data from the form field
+ *  @param form_inputs object
+ */
+function startNewGame(form_inputs) {
+    let rows = form_inputs.row
+    let columns = form_inputs.column
+    let gophers = form_inputs.gopher
+    gridDefinition(rows, columns)
+    hitGenerator(rows, columns, gophers)
+    toggleDisplay()
+    let clickedItems = document.querySelectorAll('.game-item')
```

Comment:
const

Reply:
Also unsure about this name, are they clicked items or clickable items?

Maybe something like gridTriles would be better

Reply:
done